internal/services/machinelearning/registration.go

```diff
@@ -34,10 +34,11 @@ func (r Registration) SupportedDataSources() map[string]*pluginsdk.Resource {
 // SupportedResources returns the supported Resources supported by this Service
 func (r Registration) SupportedResources() map[string]*pluginsdk.Resource {
 	return map[string]*pluginsdk.Resource{
-		"azurerm_machine_learning_compute_cluster":   resourceComputeCluster(),
-		"azurerm_machine_learning_compute_instance":  resourceComputeInstance(),
-		"azurerm_machine_learning_inference_cluster": resourceAksInferenceCluster(),
-		"azurerm_machine_learning_synapse_spark":     resourceSynapseSpark(),
-		"azurerm_machine_learning_workspace":         resourceMachineLearningWorkspace(),
+		"azurerm_machine_learning_compute_cluster":       resourceComputeCluster(),
+		"azurerm_machine_learning_compute_instance":      resourceComputeInstance(),
+		"azurerm_machine_learning_inference_cluster":     resourceAksInferenceCluster(),
+		"azurerm_machine_learning_synapse_spark":         resourceSynapseSpark(),
+		"azurerm_machine_learning_workspace":             resourceMachineLearningWorkspace(),
+		"azurerm_machine_learning_datastore_blobstorage": resourceMachineLearningDataStore(),
```

Comment:
These should be sorted alphabetically, can we also rename the resource to `azurerm_machine_learning_datastore_blob_storage` so it's consistent with other blob_storage specific resources within the provider.

```suggestion
		"azurerm_machine_learning_compute_cluster":       resourceComputeCluster(),
		"azurerm_machine_learning_compute_instance":      resourceComputeInstance(),
		"azurerm_machine_learning_datastore_blob_storage": resourceMachineLearningDataStore(),
		"azurerm_machine_learning_inference_cluster":     resourceAksInferenceCluster(),
		"azurerm_machine_learning_synapse_spark":         resourceSynapseSpark(),
		"azurerm_machine_learning_workspace":             resourceMachineLearningWorkspace(),
```